### PR TITLE
feat: Add reports API for dashboards

### LINKED ISSUE #123: Add support for reports configuration
Enabling reports on dashboards is not actually tied to the dashboard configuration / definitions, but instead has a separate API. Considering enhancing Monaco to support `/reports` endpoint at `/api/config/v1/reports?type=DASHBOARD` to allow enabling dashboard reports.

### PR DESCRIPTION
**Pull request by CruzanCaramele**
see the issue for details.
fixes #123

### CODE REVIEW

File: cmd/monaco/test-resources/integration-all-configs/project/reports/report.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+config:
+  - report: "report.json"
+
+report:
+  - name: "/projecet/dashboard/dashboard.name"
```
Comment (patrickpichler) [inline cmd/monaco/test-resources/integration-all-configs/project/reports/report.yaml:5]: You have a typo here. `projecet` instead of `project`

Comment (patrickpichler) [reply cmd/monaco/test-resources/integration-all-configs/project/reports/report.yaml:5]: I guess you still have a typo here

Comment (CruzanCaramele) [reply cmd/monaco/test-resources/integration-all-configs/project/reports/report.yaml:5]: @patrickpichler correct it now

**Review by patrickpichler — APPROVED** [review]

File: pkg/api/api.go [hunk 2]
```diff
@@ -128,6 +128,12 @@ var apiMap = map[string]apiInput{
 		apiPath: "/api/config/v1/service/requestNaming",
 	},
 
+	// Early adopter API !
+	"reports": {
+		apiPath:                      "/api/config/v1/reports",
+		propertyNameOfGetAllResponse: "values",
```
Comment (didiladi) [inline pkg/api/api.go:134]: You could remove this line, as `values` is the default. Up to you.

**Review by didiladi — APPROVED** [review]
:100: :rocket:

**Review by UnseenWizzard — APPROVED** [review]
lgtm :+1: 
as @didiladi says, the `values` could be dropped as no definition is need on a default API. 
But it doesn't hurt either

**Pull request merged**